tests/test_logging/test_logging.py

```diff
@@ -48,11 +48,11 @@ def test_logging_to_file(python_version):
     # The following regex pattern selects file paths and all numbers.
     pattern = r"(\['[A-Z]?:?[\/\\].*cfg'])|([A-Z]?:?[\/\\].*mp4)|(\d+)"
 
-    logs = re.sub(pattern, lambda m: " " * len((m.group(0))), logs)
+    logs = re.sub(pattern, lambda m: " " * len((m.group(0))), logs).replace(" ", "")
     with open(
         os.path.join(os.path.dirname(__file__), "expected.txt"), "r"
     ) as expectedfile:
         expected = re.sub(
             pattern, lambda m: " " * len((m.group(0))), expectedfile.read()
-        )
+        ).replace(" ", "")
```

Comment:
**Issue :** I _think_ it's not the right way to fix these tests. See my review comment.

Reply:
As far as I can tell, I have to add a `manim.cfg` file to the directory where `test_logging.py` lives, correct?

If that's the case, I'm honestly far more inclined to change the test than add that file. In fact, the whole point of this PR (and more that will come after it) is so that _we explicitly don't have to do that_. So if I add the `manim.cfg` file here, it will just go away (very) soon.

Reply:
The issue is that this test run a CLI invocation, so the manim.cfg have to be there, no ?

Reply:
I was under the impression that all the tests will be run with one "main" custom `manim.cfg` file, like the "user-wide" cfg file, and those tests which _absolutely_ need to have different cfg params, like the test that checks whether the cfg files cascade properly, will have folder-wide cfg files.

If my understanding is correct, we can simply set the log width to some large number, say 1024 or 2048, in the tests-wide config, and still keep our character-per-character checks for the log testing.

Reply:
Yes, we can do that as well.

Btw, for clarity purpose I used a folder wide cfg file in #335, could you check it out ?

Reply:
> If my understanding is correct, we can simply set the log width to some large number, say 1024 or 2048, in the tests-wide config, and still keep our character-per-character checks for the log testing.

This sounds like the way to go. But also, this shouldn't be part of this PR, no? Also if I understand correctly, all of the logs will have to be regenerated so that they match the new width. Yes?  Would it be OK with you two if I did all of that in a separate PR that would (hopefully) be merged quickly?

Reply:
Yes, you're right. Just fix this here, make tests pass. 

I'm convinced that we should change the way to test logs, and so we will use a JSON format in the future. See #365

Reply:
> Just fix this here, make tests pass.

I'm a little slow today. Do you mean that we should keep the `replace` call in this PR, merge it and then prepare another PR fixing the log width thing? If that's what you meant, then this PR is ready to be merged.

Reply:
If tests are working, then I'm all good.

Reply:
@leotrs @huguesdevimeux  I found out the exact reason for the expected and actual logs not matching.

`rich`, after every logger.log() call it caters to, prints the line number of the log call, like so:

`DEBUG     Read configuration files:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 config.py:  42`

Since `config.py` has been refactored, that _particular_ `logger.log()` call has been shifted from a 2 digit line number to a 3 digit line number, causing an imbalance in the whitespace, since the regex replaces all digits with whitespace since they change.

The tests won't pass if you increase the log width, they'll pass only if you change the expected log to account for this, kind of like regenerating the reference data for the graphical tests.

I would suggest replacing the current expected logs file with the log file it actually makes.

Reply:
> I would suggest replacing the current expected logs file with the log file it actually makes.

Let's listen to master test logging :D

Reply:
@Aathish04 you are _brilliant_. Just replaced the file, and as soon as the checks are done I'm merging this